src/Sarif.Multitool/Rules/SARIF2005.ProvideHelpfulToolInformation.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Json.Pointer;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class ProvideHelpfulToolInformation : SarifValidationSkimmerBase
+    {
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF2005_ProvideHelpfulToolInformation_FullDescription_Text
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        public override string Id => RuleId.ProvideHelpfulToolInformation;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideToolVersion_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideConciseToolName_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_UseNumericToolVersions_Text)
+        };
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool == null)
+            {
+                return;
+            }
```

Comment:
>} [](start = 12, length = 1)

You don't need this check because the "visitor" code will not call this method if `tool` is `null`.